tools/extract-metadata.py

```diff
@@ -160,34 +161,47 @@ def main():
 
     tree = etree.Element("xep-infos")
 
+    has_error = False
+
     for xepfile in args.xepdir.glob("xep-*.xml"):
         number = xepfile.name.split("-", 1)[1].split(".", 1)[0]
         try:
             number = str(int(number))
         except ValueError:
             continue
 
-        with xepfile.open("rb") as f:
-            parsed = open_xml(f)
-
-        tree.append(make_metadata_element(
-            number,
-            extract_xep_metadata(parsed),
-            True,
-        ))
+        try:
+            with xepfile.open("rb") as f:
+                parsed = open_xml(f)
+
+            tree.append(make_metadata_element(
+                number,
+                extract_xep_metadata(parsed),
+                True,
+            ))
+        except Exception as exc:
```

Comment:
Especially for CI it makes sense to print all errors before exiting so that you see what went wrong. I can make this more specific to do that only for XML parser errors, that probably makes sense (since the error message won’t be helpful for other errors which could occur due to incomplete XML).